src/main/java/com/mraof/minestuck/item/StructureScannerItem.java

```diff
@@ -2,84 +2,180 @@
 	{
-		levelIn.playSound(playerIn, playerIn.getX(), playerIn.getY(), playerIn.getZ(), SoundEvents.UI_BUTTON_CLICK.value(), SoundSource.AMBIENT, 0.8F, 1.3F);
-		
-		ItemStack item = playerIn.getItemInHand(handIn);
-		boolean foundItem = false;
+		ItemStack stack = pPlayer.getItemInHand(pUsedHand);
 		
-		if(fuelItem != null)
+		if(checkFuelNeeded(pPlayer, pLevel) || pPlayer.isCreative())
 		{
-			ItemStack fuelStack = new ItemStack(fuelItem.get());
+			if (stack.getDamageValue() > 0){
+				resetCharge(stack);
+			}
+			stack.getOrCreateTag().putBoolean("Powered", true);
+			pLevel.playSound(pPlayer, pPlayer.getX(), pPlayer.getY(), pPlayer.getZ(), SoundEvents.UI_BUTTON_CLICK.get(), SoundSource.AMBIENT, 0.8F, 1.3F);
 			
-			for(ItemStack invItem : playerIn.getInventory().items)
+			if (!pLevel.isClientSide)
 			{
-				if(ItemStack.isSameItem(invItem, fuelStack))
-				{
-					foundItem = true;
-					if(!levelIn.isClientSide && levelIn.getRandom().nextFloat() >= 0.95F)
-					{
-						invItem.shrink(1);
-						levelIn.playSound(playerIn, playerIn.getX(), playerIn.getY(), playerIn.getZ(), SoundEvents.BLAZE_SHOOT, SoundSource.AMBIENT, 0.4F, 2F);
-					}
-					
-					break;
-				}
+				MutableComponent message = Component.translatable("message.temple_scanner.on");
+				pPlayer.sendSystemMessage(message.withStyle(ChatFormatting.DARK_GREEN));
+			}
+		}
+		return InteractionResultHolder.success(stack);
+		
+	}
+	
+	/**
+	 * Check if the item is powered, and if it's out of battery, recharge it.
+	 * Set the location to the nearest structure, check that a structure exists, then reduce charge if fuelled.
+	 */
+	@Override
+	public void inventoryTick(ItemStack pStack, Level pLevel, Entity pEntity, int pSlotId, boolean pIsSelected)
+	{
+		super.inventoryTick(pStack, pLevel, pEntity, pSlotId, pIsSelected);
+		
+		if(pStack.hasTag() && pStack.getTag().getBoolean("Powered") && pLevel instanceof ServerLevel sLevel)
+		{
+			if (!isCharged(pStack))
+			{
+				resetCharge(pStack);
 			}
+			
+			BlockPos pos = setLocation(pEntity, sLevel);
+			
+			checkLocation(pStack, pos);
+			
+			reduceCharge(pStack, pEntity, pLevel);
+		}
+	}
+	
+	public static boolean isCharged(ItemStack stack)
+	{
+		return stack.getDamageValue() < stack.getMaxDamage();
+	}
+	
+	public void resetCharge(ItemStack pStack)
+	{
+		pStack.setDamageValue(0);
+	}
+	
+	public BlockPos setLocation(Entity pEntity, ServerLevel sLevel)
+	{
+		return sLevel.findNearestMapStructure(structure, pEntity.blockPosition(), 100, false);
+	}
+	
+	public void checkLocation(ItemStack pStack, BlockPos pos)
+	{
+		if(pos == null)
+		{
+			pStack.getTag().remove("TargetLocation");
 		} else
 		{
-			foundItem = true;
+			pStack.getTag().put("TargetLocation", NbtUtils.writeBlockPos(pos));
 		}
-		
-		if(levelIn instanceof ServerLevel level)
+	}
+	
+	/**
+	 * Check that the fuel item is set; if not, don't reduce charge.
```

Comment:
```suggestion
	 * Does nothing if this scanner type has no fuel item that it uses.
```
Clears up ambiguity on what the fuel item is set to.